cpp/include/cudf/column/column_factories.hpp

```diff
@@ -18,7 +18,6 @@
 #include <cudf/null_mask.hpp>
 #include <cudf/types.hpp>
 #include <cudf/utilities/error.hpp>
-#include <cudf/utilities/traits.hpp>
 #include "column.hpp"
 
 #include <rmm/thrust_rmm_allocator.h>
```

Comment:
And this. 
```suggestion
```

Reply:
👍

Reply:
Actually removing this one caused a build error from `rmm`.

Reply:
What error?

Reply:
```
[  8%] Building CXX object CMakeFiles/cudf.dir/src/transform/transform.cpp.o
In file included from /home/nfs/tcomer/NVIDIA/cudf/cpp/include/cudf/detail/column_factories.hpp:22:0,
                 from /home/nfs/tcomer/NVIDIA/cudf/cpp/src/transform/transform.cpp:18:
/home/nfs/tcomer/NVIDIA/cudf/cpp/include/cudf/detail/../column/column_factories.hpp:130:16: error: ‘device_vector’ in namespace ‘rmm’ does not name a template type
     const rmm::device_vector<thrust::pair<const char*, size_type>>& strings,
                ^
/home/nfs/tcomer/NVIDIA/cudf/cpp/include/cudf/detail/../column/column_factories.hpp:130:29: error: expected ‘,’ or ‘...’ before ‘<’ token
     const rmm::device_vector<thrust::pair<const char*, size_type>>& strings,
                             ^
/home/nfs/tcomer/NVIDIA/cudf/cpp/include/cudf/detail/../column/column_factories.hpp:168:16: error: ‘device_vector’ in namespace ‘rmm’ does not name a template type
     const rmm::device_vector<char>& strings,
                ^
/home/nfs/tcomer/NVIDIA/cudf/cpp/include/cudf/detail/../column/column_factories.hpp:168:29: error: expected ‘,’ or ‘...’ before ‘<’ token
     const rmm::device_vector<char>& strings,
                             ^
CMakeFiles/cudf.dir/build.make:504: recipe for target 'CMakeFiles/cudf.dir/src/transform/transform.cpp.o' failed
make[2]: *** [CMakeFiles/cudf.dir/src/transform/transform.cpp.o] Error 1
```

Reply:
Does it work if you forward declare device vector?